ghost/core/core/server/data/migrations/versions/5.21/2022-10-26-13-32-backfill-subscriptions.js

```diff
@@ -0,0 +1,112 @@
+            logging.warn(`Unable to fetch free product - skipping subscription backfill`);
+            return;
+        }
+
+        const now = knex.raw('CURRENT_TIMESTAMP');
+
+        const ghostSubscriptionIds = [];
+
+        const rowsToInsert = members.map((m) => {
+            const subscriptionId = ObjectId().toHexString();
+
+            const row = {
+                id: subscriptionId,
+                type: m.status,
+                member_id: m.id,
+                created_at: now
+            };
+
+            if (m.status === 'free') {
+                row.status = 'active';
+                row.tier_id = freeProduct.id;
+            } else if (m.status === 'comped') {
+                row.status = 'active';
+                row.tier_id = m.product_id;
+                row.expires_at = m.expiry_at;
+            } else if (m.status === 'paid') {
+                row.status = 'active';
+                row.tier_id = m.product_id;
+                row.offer_id = m.offer_id;
+
+                ghostSubscriptionIds.push({
+                    ghost_subscription_id: subscriptionId,
+                    mscs_id: m.mscs_id
+                });
+            }
+
+            return row;
+        });
+
+        logging.info(`Inserting ${rowsToInsert.length} backfilled subscriptions`);
+        await knex.batchInsert('subscriptions', rowsToInsert);
+
+        if (ghostSubscriptionIds.length) {
+            const batches = _.chunk(ghostSubscriptionIds, 1000);
+            logging.info(`Backfilling ${ghostSubscriptionIds.length} Ghost subscription IDs into members_stripe_customers_subscriptions table via ${batches.length} batches`);
+
+            for (const batch of batches) { // eslint-disable-line no-restricted-syntax
+                const batchResponse = await knex('members_stripe_customers_subscriptions').insert(batch.map(r => ({
+                    id: r.mscs_id,
+                    ghost_subscription_id: r.ghost_subscription_id,
+
+                    created_at: knex.raw('NOW()'),
+                    created_by: '',
+                    customer_id: '',
+                    current_period_end: knex.raw('NOW()'),
+                    plan_amount: 0,
+                    plan_currency: '',
+                    plan_id: '',
+                    plan_interval: '',
+                    plan_nickname: '',
+                    start_date: knex.raw('NOW()'),
+                    status: '',
+                    subscription_id: ''
+                }))).onConflict('id').merge(['ghost_subscription_id']);
+
+                if (batchResponse[0] !== 0) {
+                    logging.error(`Inserted ${batchResponse[0]} members_stripe_customers_subscriptions, expected 0`);
+                    throw new Error('Rolling back');
+                }
+            }
+        } else {
+            logging.info(`No new Ghost subscriptions need backfilling into members_stripe_customers_subscriptions`);
+        }
+    },
+    async function down(knex) {
```

Comment:
This needs changing/thinking about, but it's good for testing because it resets us back